tests/TileDB.CSharp.Test/FragmentInfoTest.cs

```diff
@@ -0,0 +1,511 @@
+
+                var dirLength = (ulong)TestUtil.GetDirectorySize(fragmentUri.LocalPath);
+
+                Assert.AreEqual(dirLength, info.GetFragmentSize(i));
+            }
+        }
+
+        [DataTestMethod]
+        [DataRow(true, DisplayName = "Dense")]
+        [DataRow(false, DisplayName = "Sparse")]
+        public void TestIsDenseSparse(bool isDense)
+        {
+            using var uri = new TemporaryDirectory("fragment_info_is_dense_sparse");
+
+            CreateArray(uri, isDense);
+            for (uint i = 0; i < FragmentCount; i++)
+            {
+                WriteArray(uri, isDense);
+            }
+
+            using FragmentInfo info = new FragmentInfo(_ctx, uri);
+            info.Load();
+
+            uint fragmentCount = info.FragmentCount;
+
+            for (uint i = 0; i < fragmentCount; i++)
+            {
+                Assert.AreEqual(isDense, info.IsDense(i));
+                Assert.AreEqual(!isDense, info.IsSparse(i));
+            }
+        }
+
+        [DataTestMethod]
+        [DataRow(true, DisplayName = "Dense")]
+        [DataRow(false, DisplayName = "Sparse")]
+        public void TestGetTimestampRange(bool isDense)
+        {
+            using var uri = new TemporaryDirectory("fragment_info_timestamp_range");
+
+            CreateArray(uri, isDense);
+
+            for (uint i = 0; i < FragmentCount; i++)
+            {
+                WriteArray(uri, isDense);
+            }
+
+            using FragmentInfo info = new FragmentInfo(_ctx, uri);
+            info.Load();
+
+            uint fragmentCount = info.FragmentCount;
+
+            for (uint i = 0; i < fragmentCount; i++)
+            {
+                Uri fragmentUri = new(info.GetFragmentUri(i));
+
+                (ulong, ulong) expectedTimestampRange = info.GetTimestampRange(i);
+
+                string[] dirNameSplit = Path.GetFileName(fragmentUri.LocalPath)!.Split('_');
+
+                ulong start = ulong.Parse(dirNameSplit[2]);
+                ulong end = ulong.Parse(dirNameSplit[3]);
+
+                Assert.AreEqual(expectedTimestampRange, (start, end));
+            }
+        }
+
+        [TestMethod]
+        public void TestMinimumBoundedRectanglesString()
+        {
+            using var uri = new TemporaryDirectory("fragment_info_mbr_var");
+            CreateSparseVarDimArrayForMbrTesting(uri);
+            WriteSparseVarDimArrayForMbrTesting(uri);
+
+            using FragmentInfo info = new FragmentInfo(_ctx, uri);
+            info.Load();
+
+            Assert.AreEqual(1u, info.FragmentCount);
+
+            Assert.AreEqual(2ul, info.GetMinimumBoundedRectangleCount(0));
+
+            Assert.AreEqual(("a", "bb"), info.GetMinimumBoundedRectangle<string>(0, 0, 0));
+            Assert.AreEqual(("c", "ddd"), info.GetMinimumBoundedRectangle<string>(0, 1, "d"));
+        }
+
+        [TestMethod]
+        public void TestMinimumBoundedRectangles()
```

Comment:
Can we parametrize this over different numeric types?

Reply:
I copy-pasted the MBR test to work with attributes with 32-bit integers. We could avoid duplication with [.NET 7's generic math](https://learn.microsoft.com/en-us/dotnet/standard/generics/math) but until then…